GameTracker-Core/DirectorySearchHelper.cs

```diff
@@ -0,0 +1,83 @@
+﻿using GameTracker_Core.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace GameTracker_Core
+{
+    public static class DirectorySearchHelper
+    {
+        public static bool DirectoryExecuteSearch(string dir)
+        {
+            return DirectoryExecuteSearchRecursive(dir, false);
+        }
+
+        private static bool DirectoryExecuteSearchRecursive(string dir, bool erg)
+        {
+            try
```

Comment:
You could return `true` here if `erg` is already `true` to avoid unnecessary overhead.